Packs/LogRhythmRest/Integrations/LogRhythmRest/LogRhythmRest.py

```diff
@@ -583,6 +1580,200 @@ def get_alarm_events(data_args):
+        query.append(generate_query_item(filterType=136, valueType=2, value=int(ENTITY_ID)))
+
+    if source_type and source_type != "all":
+        query.append(generate_query_item(filterType=9, valueType=2, value=SOURCE_TYPE_MAP[source_type]))
+
+    if username:
+        query.append(generate_query_item(filterType=43, valueType=4, value=str(username)))
+
+    if subject:
+        query.append(generate_query_item(filterType=33, valueType=4, value=str(subject)))
+
+    if sender:
+        query.append(generate_query_item(filterType=31, valueType=4, value=str(sender)))
+
+    if recipient:
+        query.append(generate_query_item(filterType=32, valueType=4, value=str(recipient)))
+
+    if hash:
+        query.append(generate_query_item(filterType=138, valueType=4, value=str(hash)))
+
+    if url:
+        query.append(generate_query_item(filterType=42, valueType=4, value=str(url)))
+
+    if process_name:
+        query.append(generate_query_item(filterType=41, valueType=4, value=str(process_name)))
+
+    if object:
+        query.append(generate_query_item(filterType=34, valueType=4, value=str(object)))
+
+    if ipaddress:
+        query.append(generate_query_item(filterType=17, valueType=5, value=str(ipaddress)))
+
+    # Search and get TaskID
+    querybody = {
+        "maxMsgsToQuery": 50,
+        "logCacheSize": 10000,
+        "queryTimeout": 60,
+        "queryRawLog": True,
+        "queryEventManager": False,
+        "dateCriteria": {
+            "useInsertedDate": False,
+            "lastIntervalValue": int(number_of_date),
+            "lastIntervalUnit": 4
+        },
+        "queryLogSources": [],
+        "queryFilter": {
+            "msgFilterType": 2,
+            "isSavedFilter": False,
+            "filterGroup": {
+                "filterItemType": 1,
+                "fieldOperator": 1,
+                "filterMode": 1,
+                "filterGroupOperator": 0,
+                "filterItems": query
+            }
+        }
+    }
+
+    headers = dict(HEADERS)
+    headers['Content-Type'] = 'application/json'
+
+    search_task = http_request('POST', 'lr-search-api/actions/search-task', json.dumps(querybody), headers)
+    task_id = search_task["TaskId"]
+
+    results = CommandResults(
+        outputs={"taskID": task_id},
+        outputs_prefix="Logrhythm.SearchResult"
+    )
+
+    return_results(results)
+
+
+def lr_get_query_result(data_args):
+    task_id = data_args.get('task_id')
+
+    queryresult = json.dumps(
+        {
+            "data": {
+                "searchGuid": task_id,
+                "search": {
+                    "sort": [],
+                    "fields": []
+                },
+                "paginator": {
+                    "origin": 0,
+                    "page_size": 50
+                }
+            }
+        })
+
+    headers = dict(HEADERS)
+    headers['Content-Type'] = 'application/json'
+
+    run = 1
+    while True:
```

Comment:
remove the while statement, the code inside should be executed once

Reply:
The search result will not be returned immediately. Normally it took about 5-30 secs depending on the size of the search and LogRhythm performance, so I put the while to check for the result every 5 seconds. If Log Rhythm has the result, it will return and stop. 
Do you have any idea to optimize this method?